functions/slackEvents.js

```diff
@@ -139,10 +140,33 @@ async function onMessageEvent(event) {
 }
 
 async function handleAction(action, user) {
+  const [volunteerKey, volunteer] = await getVolunteerBySlackID(user.id);
+  if (!volunteer) {
+    throw new Error(`No volunteer found for slack id ${user.id}`);
+  }
+
   switch (action.action_id) {
   case 'email_shopping_list':
     try {
-      await emailShoppingList(action.value, user.id);
+      const records = await getRecordsWithTicketID(INTAKE_TABLE, action.value);
+      validateVolunteerAssigned(records, volunteerKey);
+      const email = await getSingleShoppingListEmail(records, volunteer);
+      email.send();
+    } catch (err) {
+      console.error('Failed to email shopping list to user', { user: user.id, err: err });
+      return 1;
+    }
+
+    return 0;
+  case 'email_consolidated_shopping_list':
+    try {
+      const records = await getRecordsWithFilter(INTAKE_TABLE, {
+        status: 'Assigned / In Progress',
+        deliveryVolunteer: volunteerKey
+      });
+      validateVolunteerAssigned(records, volunteerKey);
```

Comment:
is this necessary here if that's how you're filtering the tickets from airtable

Reply:
Probably not, just paranoid.